viewing_party/party.py

```diff
@@ -1,23 +1,276 @@
 # ------------- WAVE 1 --------------------
+from collections import Counter
+from enum import unique
 
+# inputs a movie into the program
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie = {
+            "title" : title,
+            "genre" : genre,
+            "rating" : rating
+        }
+        return movie
+    else:
+        return None
+
+# takes a movie dict and adds it to user's watched
+def add_to_watched(user_data, movie):
+    user_data["watched"] = [movie]
+    return user_data
+
+# takes a movie dict and adds it to user's watchlist
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+    return user_data
+
+# finds the title of a movie in watchlist, then removes and transfers it to watched
+def watch_movie(user_data, title):
+    counter = 0
+    for movie in user_data["watchlist"]:
+        if title in movie["title"]:
+            user_data["watched"].append(movie)
+            del(user_data["watchlist"][counter])
+        # if title in user_data["watchlist"]:
+            # how to remove the title from the watchlist
+            # movie_seen = user_data["watchlist"].pop(title)
+            # movie_seen = user_data.pop("watchlist")
+            # user_data["watched"][movie] = [movie_seen]
+            return user_data
+        counter += 1  
+    return user_data     
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# calculates the average rating of every movie the user has watched
+def get_watched_avg_rating(user_data):
+    rating = 0
+    for movie in user_data["watched"]:
+        rating += movie["rating"]
+    if rating == 0:
+        return 0
+    else:
+        avg_rating = rating / len(user_data["watched"])
+        return avg_rating
+
+# finds the user's most frequently watched genre
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0:

```

Comment:
It would be more "Pythonic" to use 
```python
if not user_data["watched"]:
``` 
This works because `user_data["watched"]` returns an empty list, which is falsy.